metakb/transform/civic.py

```diff
@@ -25,553 +68,519 @@ def __init__(self,
+            variant_query = f"{variant['entrez_name']} {variant_name}"
+
+            if not self._is_supported_variant_query(variant_name, variant_id):
                 continue
 
-            variation_descriptor = await self.vicc_normalizers.normalize_variation(
+            vrs_variation = await self.vicc_normalizers.normalize_variation(
                 [variant_query]
             )
 
             # Couldn't find normalized concept
-            if not variation_descriptor:
-                logger.warning("Variation Normalizer unable to normalize "
-                               f"civic.vid:{variant['id']} using query "
-                               f"{variant_query}")
+            if not vrs_variation:
+                logger.debug("Variation Normalizer unable to normalize "
+                             f"{variant_id} using query {variant_query}")
                 continue
 
-            if variant["variant_types"]:
-                structural_type = variant["variant_types"][0]["so_id"]
-            else:
-                structural_type = None
-
-            variation_descriptor = VariationDescriptor(
-                id=variant_id,
-                label=variant["name"],
-                description=variant["description"] if variant["description"] else None,  # noqa: E501
-                variation_id=variation_descriptor.variation_id,
-                variation=variation_descriptor.variation,
-                gene_context=f"civic.gid:{variant['gene_id']}",
-                structural_type=structural_type,
-                expressions=hgvs_exprs,
-                xrefs=self._get_variant_xrefs(variant),
-                alternate_labels=[v_alias for v_alias in
-                                  variant["variant_aliases"] if not
-                                  v_alias.startswith("RS")],
-                extensions=self._get_variant_extensions(variant)
-            ).dict(by_alias=True, exclude_none=True)
-            self.valid_ids["variation_descriptors"][variant_id] = \
-                variation_descriptor
-            self.variation_descriptors.append(
-                variation_descriptor
-            )
-
-    def _get_variant_extensions(self, variant) -> list:
-        """Return a list of extensions for a variant.
-
-        :param dict variant: A CIViC variant record
-        :return: A list of extensions
-        """
-        extensions = [
-            Extension(
-                name='civic_representative_coordinate',
-                value={k: v for k, v in variant['coordinates'].items()
-                       if v is not None}
-            ).dict(exclude_none=True),
-            Extension(
-                name='civic_actionability_score',
-                value=variant['civic_actionability_score']
-            ).dict(exclude_none=True)
-        ]
+            # Create VRS Variation object
+            params = vrs_variation.model_dump(exclude_none=True)
+            params["id"] = vrs_variation.id
```

Comment:
is this line needed? shouldn't `params` already have this field?

Reply:
Right, thanks!